cli/daemon/httpserver.go

```diff
@@ -0,0 +1,130 @@
+package daemon
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/tarantool/tt/cli/daemon/api"
+	"github.com/tarantool/tt/cli/ttlog"
+)
+
+const (
+	defaultStopTimeout = 100 * time.Millisecond
+)
+
+type HTTPServer struct {
+	listenIface string
+	port        int
+	srv         *http.Server
+	timeout     time.Duration
+	logger      *ttlog.Logger
+}
+
+func (httpServer *HTTPServer) externalIP() (string, error) {
+	if httpServer.listenIface == "" {
+		return net.IPv4zero.String(), nil
+	}
+
+	iface, err := net.InterfaceByName(httpServer.listenIface)
+	if err != nil {
+		return "", err
+	}
+
+	if iface.Flags&net.FlagUp == 0 {
+		return "", fmt.Errorf("Interface down")
+	}
+	if iface.Flags&net.FlagLoopback != 0 {
+		return "", fmt.Errorf("Loopback interface")
+	}
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
Add a comment that the socket will bind to the first discovered IP address on the specified interface.

Reply:
Added.